go/arrow/rle/rle_utils_test.go

```diff
@@ -0,0 +1,144 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rle_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/apache/arrow/go/v10/arrow"
+	"github.com/apache/arrow/go/v10/arrow/array"
+	"github.com/apache/arrow/go/v10/arrow/memory"
+	"github.com/apache/arrow/go/v10/arrow/rle"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFindPhysicalOffset(t *testing.T) {
+	tests := []struct {
+		vals []int32
+		find int
+		exp  int
+	}{
+		{[]int32{1}, 0, 0},
+		{[]int32{1, 2, 3}, 0, 0},
+		{[]int32{1, 2, 3}, 1, 1},
+		{[]int32{1, 2, 3}, 2, 2},
+		{[]int32{2, 3, 4}, 0, 0},
+		{[]int32{2, 3, 4}, 1, 0},
+		{[]int32{2, 3, 4}, 2, 1},
+		{[]int32{2, 3, 4}, 3, 2},
+		{[]int32{2, 4, 6}, 3, 1},
+		{[]int32{1, 2, 3, 4, 5, 6, 7, 8, 9, 1000, 1005, 1015, 1020, 1025, 1050}, 1000, 10},
+		// out-of-range logical offset should return len(vals)
+		{[]int32{2, 4, 6}, 6, 3},
+		{[]int32{2, 4, 6}, 10000, 3},
```

Comment:
What about out-of-range in the negative direction?  Error?

Reply:
I'll add a test case for it, I originally just copied these tests over from the C++ draft PR :smile:

Reply:
Hmm, what do we think *should* happen? going out of range in the positive direction just returns `len(vals)` technically the offset should never be negative, rather than error I think the better solution here would be to just make the argument a `uint` rather than an `int`, thoughts?